enumerables.rb

```diff
@@ -0,0 +1,108 @@
+module Enumerable
+  def my_each
+    length.times { |i| yield(self[i]) }
+  end
+
+  def my_each_with_index
+    (0..length - 1).each do |index|
+      yield(self[index], index)
+    end
+  end
+
+  def my_select
+    result = []
+    my_each do |item|
+      result << item if yield(item)
+    end
+    result
+  end
+
+  def my_all?
+    result = true
+    my_each { |i| break result = false unless yield(i) }
+    result
+  end
+
+  def my_any?
+    result = false
+    my_each { |i| break result = true if yield(i) }
+    result
+  end
+
+  def my_none?
+    result = true
+    my_each { |i| break result = false if yield(i) }
+    result
+  end
```

Comment:
* [ ] should have working `my_none?` identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1, false, 'hi', []].my_none? != [1, false, 'hi', []].none?

[1,2,3,4,5].my_none?(String) != [1,2,3,4,5].none?(String)

['sure', 'trust'].my_none?(/z/) != ['sure', 'trust'].none?(/z/)

['sure', 'trust'].my_none?(5) != ['sure', 'trust'].none?(5)
```